include/silo/common/numbers.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <cstdint>
+
+/// Throws `std::overflow_error` if `val` isn't in uint32_t range.
+uint32_t uint64ToUint32(uint64_t val);
+
+/// Throws `std::overflow_error` on overflow
+uint32_t inc(uint32_t val);
```

Comment:
It's interesting how C++ can make one think this because it actually allows writing such functions.

Why is `safeAddOne` making you think that less than `inc`?

What about `safeInc`?

Reply:
I thought of `inc` as `val++` with an additional bounds check (when I didn't read the code careful enough initially).

"add one" makes me assume that I will get a return value, whereas "inc" made me assume that it happens in place.

Reply:
Maybe because "++" is called increment, I could see that. I've been used to `inc` functions in functional languages. I've renamed it to `add1` now.